docs/design/HealthEndpoint.md

```diff
@@ -0,0 +1,206 @@
+| `enabled`         | Boolean       | No           | `true`      | Enables or disables health checks for the data source.                                              |
+| `query`           | String        | No           | N/A         | Custom SQL query used to perform the health check.                                                  |
+| `threshold-ms`    | Integer       | No           | `10000`     | Threshold in milliseconds for the query response time before the check is considered degraded.      |
+
+#### `<entity-name>.health` Configuration
+
+The Entity config parameters contain information about the GET filter or query which needs to be carried out on the data source entiy and under what threshold should the response be received for it to qualify as a healthy entity.
+| **Property**     | **Data Type** | **Required** | **Default** | **Description**                                                                                      |
+|------------------|---------------|--------------|-------------|------------------------------------------------------------------------------------------------------|
+| `enabled`        | Boolean       | No           | `true`      | Enables or disables health checks for the specific entity.                                          |
+| `filter`         | String        | No           | `null`      | Filter condition applied to the health check query (e.g., `"Id eq 1"`).                             |
+| `first`          | Integer       | No           | `1`         | Number of records to query during the health check.                                                 |
+| `threshold-ms`   | Integer       | No           | `10000`     | Threshold in milliseconds for the query response time before the check is considered degraded.      |
+
+#### Example
+```
+{
+  "runtime" : {
+    "health" : {
+      "enabled": true, (default: true)
+      "cache-ttl": 5, (optional default: 5)
+      "max-dop": 5, (optional default: 1)
+      "roles": ["anonymous", "authenticated"] (optional default: *)
+    }
+  }
+}
+{
+  "data-source" : {
+    "health" : {
+      "moniker": "sqlserver", (optional default: NULL) // not required, on purpose, most have just one
+      "enabled": true, (default: true)
+      "query": "SELECT TOP 1 1", (option)
+      "threshold-ms": 100 (optional default: 10000)
+    }
+  }
+}
+{
+  "<entity-name>": {
+      "health": {
+        "enabled": true, (default: true)
+        "filter": "Id eq 1" (optional default: null),
+        "first": 1 (optional default: 1),
+        "threshold-ms": 100 (optional default: 10000)
+      },
+      ...
+    },
+  }
+}
+```
+The idea of using this updates configuration is to allow the developer to influence how the checks work against the datasource/entity. This would provide him with a more detailed process for checking if DAB engine is healthy and would give him an enhanced user experience. 
```

Comment:
```suggestion
The idea of using this updates configuration is to allow the developer to influence how the checks work against the datasource/entity. This would provide them with a more detailed process for checking if DAB engine is healthy and would give them an enhanced user experience. 
```
Being more neutral on the use of pronouns.